File: app/src/main/java/com/seniorhub/config/CloudinaryConfig.kt
package com.seniorhub.config

/**
 * Cloudinary Configuration for SeniorHub Android App
 * 
 * This configuration matches the admin web interface settings
 * to ensure consistent image handling across all platforms.
 */
object CloudinaryConfig {
    
    // Cloudinary credentials - Replace with your actual values
    const val CLOUD_NAME = "dftnlg6nh"
    const val API_KEY = "578778791638823"
    const val API_SECRET = "o9ZTgXlca2aqlMnX8t17Rj75D7M"
    
    // Upload settings
    const val UPLOAD_PRESET = "senior-hub-upload"
    const val FOLDER = "senior-hub/profiles"
    
    // Image transformation settings
    object Transformations {
        // Profile image transformations
        const val PROFILE = "w_400,h_400,c_fill,g_face,q_auto,f_auto"
        const val THUMBNAIL = "w_100,h_100,c_fill,g_face,q_auto,f_auto"
        const val FULL_SIZE = "w_800,h_800,c_fill,g_face,q_auto,f_auto"
        
        // Admin display transformations
        const val ADMIN_LIST = "w_60,h_60,c_fill,g_face,q_auto,f_auto"
        const val ADMIN_DETAIL = "w_200,h_200,c_fill,g_face,q_auto,f_auto"
    }
    
    // File validation settings
    object Validation {
        const val MAX_FILE_SIZE = 5 * 1024 * 1024 // 5MB
        val ALLOWED_TYPES = listOf(
            "image/jpeg",
            "image/png", 
            "image/webp",
            "image/gif"
        )
    }
    
    // URL generation helpers
    fun getProfileImageUrl(publicId: String, transformation: String = Transformations.PROFILE): String {
        return "https://res.cloudinary.com/$CLOUD_NAME/image/upload/$transformation/$publicId"
    }
    
    fun getAdminImageUrl(publicId: String, transformation: String = Transformations.ADMIN_LIST): String {
        return "https://res.cloudinary.com/$CLOUD_NAME/image/upload/$transformation/$publicId"
    }
}

File: admin-web/js/cloudinary-config.js
// Cloudinary Configuration
// Replace these values with your actual Cloudinary credentials

const CLOUDINARY_CONFIG = {
    // Your Cloudinary cloud name (found in your Cloudinary dashboard)
    cloudName: 'dftnlg6nh',
    
    // Upload presets to try (in order of preference)
    uploadPresets: [
        'senior-hub-upload',    // This one is working! Use it first
        'ml_default',           // Try default preset
    ],
    
    // Current upload preset (will be set dynamically)
    uploadPreset: 'senior-hub-upload',
    
    // Optional: API key and secret (for server-side operations)
    apiKey: '578778791638823',
    apiSecret: 'o9ZTgXlca2aqlMnX8t17Rj75D7M'
};

// Image upload settings
const UPLOAD_SETTINGS = {
    // Folder to organize images
    folder: 'senior-hub/profiles',
    
    // Image transformations - DISABLED for unsigned uploads
    // transformations: {
    //     // Resize to 400x400, crop to face, auto-format
    //     profile: 'w_400,h_400,c_fill,g_face,f_auto',
    //     
    //     // Thumbnail for lists
    //     thumbnail: 'w_100,h_100,c_fill,g_face,f_auto',
    //     
    //     // Full size for detailed view
    //     full: 'w_800,h_800,c_fill,g_face,f_auto'
    // },
    
    // File size limits
    maxFileSize: 5 * 1024 * 1024, // 5MB
    
    // Allowed file types
    allowedTypes: ['image/jpeg', 'image/png', 'image/webp', 'image/gif']
};

// Export for use in other files
window.CLOUDINARY_CONFIG = CLOUDINARY_CONFIG;
window.UPLOAD_SETTINGS = UPLOAD_SETTINGS;
